src/SarifCli/AnalyzeCommand.cs

```diff
@@ -0,0 +1,148 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Microsoft.CodeAnalysis.Sarif.Driver;
+using Microsoft.Json.Schema;
+using Microsoft.Json.Schema.Validation;
+
+namespace Microsoft.CodeAnalysis.Sarif.Cli
+{
+    internal class AnalyzeCommand : AnalyzeCommandBase<SarifValidationContext, AnalyzeOptions>
+    {
+        public override string Prerelease => VersionConstants.Prerelease;
+
+        private List<Assembly> _defaultPlugInAssemblies;
+
+        public override IEnumerable<Assembly> DefaultPlugInAssemblies
+        {
+            get
+            {
+                if (_defaultPlugInAssemblies == null)
+                {
+                    _defaultPlugInAssemblies = new List<Assembly>
+                    {
+                        this.GetType().Assembly
+                    };
+                }
+
+                return _defaultPlugInAssemblies;
+            }
+        }
+
+        protected override SarifValidationContext CreateContext(AnalyzeOptions options, IAnalysisLogger logger, RuntimeConditions runtimeErrors, string filePath = null)
+        {
+            SarifValidationContext context = base.CreateContext(options, logger, runtimeErrors, filePath);
+            context.SchemaFilePath = options.SchemaFilePath;
+            return context;
+        }
+
+        protected override void AnalyzeTarget(IEnumerable<ISkimmer<SarifValidationContext>> skimmers, SarifValidationContext context, HashSet<string> disabledSkimmers)
+        {
+            // The base class knows how to invoke the skimmers that implement smart validation...
+            base.AnalyzeTarget(skimmers, context, disabledSkimmers);
+
+            // ... but it doesn't know how to invoke schema validation, which has its own set of rules,
+            // so we do that ourselves.
+            Validate(context.TargetUri.LocalPath, context.SchemaFilePath, context.Logger);
+        }
+
+        private void Validate(string instanceFilePath, string schemaFilePath, IAnalysisLogger logger)
+        {
+            try
+            {
+                string instanceText = File.ReadAllText(instanceFilePath);
+                PerformSchemaValidation(instanceText, instanceFilePath, schemaFilePath, logger);
+            }
+            catch (JsonSyntaxException ex)
+            {
+                ReportResult(ex.Result, logger);
+            }
+            catch (SchemaValidationException ex)
+            {
+                ReportInvalidSchemaErrors(ex, schemaFilePath, logger);
+            }
+            catch (Exception ex)
```

Comment:
You're right. Fixed. Removed the handler.

---
In reply to: [77238313](https://github.com/Microsoft/sarif-sdk/pull/501#discussion_r77238313) [](ancestors = 77238313)